### ISSUE TITLE
Force Fit New Source Back in Time

### ISSUE THREAD

**Issue by ajstewart** [transientskp/tkp#419]
When a new source is found it would be great if that position could be measured in any previous epochs as well as the ones going forward.

I had a look through and couldn't see this feature request, if I missed it then feel free to delete.

**Comment by AntoniaR** [comment]
Doing this is a deliberate design decision within TraP as you very rapidly hit performance problems by going back through previous images and we cannot assume the previous images are available (see TraP paper for full discussion and the upcoming Hobbs et al. where we compare TraP to another pipeline that does do this). I maintain that this is still the correct strategy and does not hinder transient identification. If you do want the flux points from the previous images, there are two methods available:

1. Run pyse standalone on all the previous images to grab the fluxes

2. Rerun TraP with the sources of interest in the monitoring list

Option 1 is likely faster if you just want the datapoints for plotting as you can parallelise it more.

**Comment by timstaley** [comment]
Hmm. I think I'm with Antonia on this one. There's no hard reason we couldn't have an 'offline' mode that is significantly slower, exactly - but the implementation could be problematic. Because we record variability indices at each time-step, we'd have to go back and update these, and also the positions might shift, so then your associations change, and... I feel a disturbance in the codebase, as a hundred developer-hours cry out in pain and terror. 
As Antonia points out, at the cost of extra processing time (but no dev-time - yay!) you can get around this with a monitoringlist position, which is already implemented in such a fashion as to be safely isolated from other sources. Now... if you wanted a script that pulls out significant transient detections from the first TraP run, then re-runs with these as monitoringlist positions - I'd be happy to help with that.

**Comment by ajstewart** [comment]
Ok, I agree with the structure actually. It makes sense and I appreciate how complex it can quickly become. And with how the TraP is designed - it can't record something from the off that it didn't know was going to be there in the first place, that certainly doesn't make sense.

I think at the time I posted this I was struck by a plot in a paper I saw of the TraP light curve along with lots of other light curves (I think you generated that Antonia - I can't remember what it was for something to do with ASKAP?) where it was the only one which was not 'complete'. That and Rob always pointing it out to me whenever I show him a light curve from the TraP, which I can understand why he would.

As you say the capability is there, especially with the monitoring list, I think perhaps it needs to be made easier though. More so for any potential new users. Maybe all this needs is a 'generate monitoring list file' button on Banana - then it's easy to just re-run. Just to save writing an external script to extract the relevant info which is sitting on the screen in front of you. For me - as I'm using Antonia's scripts I have all this info quite easily so I will be ok in generating a list.

As a side note - perhaps the monitoring list feature needs to be highlighted more in the docs? I have never used it and did not know how and it took me a good while to find a mention of it in the docs. For example I went straight to pipeline configuration, and while now I understand why it's not there, to me that's the logical first place to look when trying to find out how to use the feature.

**Comment by AntoniaR** [comment]
Yes, the plot I have shown in the past is from an ASKAP/BETA publication where there are multiple light curves from:
1. Measuring the flux of a single pixel at position of source
2. TraP blind detection (i.e. only after first detection)
3. TraP monitoring list (complete light curve)
4. VAST (which currently goes back and fits for whole light curve)
In that paper, we carefully explain why the TraP blind light curve is "incomplete" and why it is the correct behaviour for a pipeline that needs to run in real-time with a continuous stream of images. That paper is submitted but not on arxiv yet, but I have shared the plot at a pipeline meeting.

Yes, I agree that the monitoring list could be highlighted more and potentially have a way of automatically generating a monitoring list file. If this is put into Banana, I would like to see users being able to select which sources they want - rather than a file that would monitor every single source as to me that seems excessive for most scenarios. You should be able to tell from the light curve that is on Banana if they are an interesting source - where a full light curve is needed for further analysis or publication - or sources that are simply close to the detection threshold.

**Comment by ajstewart** [comment]
Ah great! Sorry the last time I saw the plot it was flashed up in a talk. I didn't realise the next one was the monitoring list.

And yes I'd agree with the source selection in generating a monitoring list file. That would be the ideal method I think. I'll close this issue and perhaps create a low priority enhancement for some sort of auto monitoring list generation.
